cql3/expr/expression.cc

```diff
@@ -903,6 +903,29 @@ expression replace_column_def(const expression& expr, const column_definition* n
         }, expr);
 }
 
+// Replaces all occurences of token(p1, p2) on the left hand side with the given colum.
```

Comment:
I don't understand the focus on "left hand side". This is fine for the expression `token(p1) < token(1)`, but what about the expression `token(1) > token(p1)`? Isn't this a legal expression in CQL? If it is NOT legal, please mention this in the comment and write a test to demonstrate that it's not legal.

Reply:
CQL expressions are now asymmetric (I want to change that). The token function can only go into the left hand side, and constants/bind variables can only go into the right hand side.

Reply:
> `token(1) > token(p1)`? Isn't this a legal expression in CQL?

No, just like `3=col` is not legal.

> If it is NOT legal, please mention this in the comment and write a test to demonstrate that it's not legal.

I don't think this is necessary.  We don't need tests that ensure CQL stays as restrictive tomorrow as it is today.  Such tests have proven nothing but a hindrance so far.

And it's fine to presume the reader is familiar with CQL (rather than peppering the code with unconnected and unsynchronized explanation fragments), just like it's fine to presume they are familiar with C++ and don't need an explanation of, eg, prvalues.

Reply:
Clearly, I'm not worried that CQL "stays as restrictive tomorrow as it is today". I'm more worried whether it is in fact restrictive in the same way in Cassandra as it is in Scylla and this asymmetry isn't just a Scylla bug.
Moreover, there is also value of having a large number of short tests which highlight various corner cases that developers might not be aware of. I'm not a CQL newbie, and only today I realized that token(1) > token(p) or 3=col is not legal, I don't think it's obvious that it's not legal. 
Finally, tests serve as a way to catch **regressions**. Consider in the future someone changes the code so the reverse order *is* supported. If we have no tests for the reverse order, it may start working but work incorrectly - and nobody will know. If we have a test that says that reverse order is forbidden, the developer is forced to remove this test - and hopefully remember to replace it with a tests for the reverse order.

@dekimir if you want to convince me of your last paragraph, I suggest you do a poll and ask Scylla developers how many know for a fact (and not just a guess - like I was able to guess) that while "token(p) < 0" is legal CQL, "0 > token(p)" is not. I claim it's not obvious at all. And I genuinely worry that one day we'll lift this restriction but then forget that we have code lying around which handles the "left hand side" of expressions in a special way.

Reply:
I'm resolving this discussion because I don't think we need to anything about it in the context of this PR, but I think it's definitely an important point we'll need to address in the future: If in the future we'll allow "reversed" expressions (with the variable on the right hand side) we'll need to change a lot of the tests, because we could have "left-hand-side"-specific code in a lot of places, not just in this new token code.